### PR TITLE
feat: add `Add to list` dropdown to user profile

### LINKED ISSUE #1828: Feature: replace "Get Card" CTA for "Add to List" dropdown
### Type of feature

🍕 Feature

### Current behavior

These are the current CTAs in the header component on User Profile pages:
image

### Suggested solution

We want to replace the current "Get Card" CTA with a new "Add to list" CTA. 
- This CTA would trigger a dropdown where the user can see the lists they have and add the user to a list.
- The dropdown should be a multiselect dropdown, where the user can select more than one list

There is a new design for the "Get Card" CTA, but as we ship that, we can move the "Get Card" option to the three dot menu

image

If the user has no lists, show a "You have no lists" message in the dropdown and add a link to the "New List" page.

image

### Update on "Get Card" CTA
We are moving the action to a new icon button  -- see issue #1829 

### Additional context

_No response_

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [X] I agree to follow this project's Contribution Docs

### PR DESCRIPTION
**Pull request by OgDev-01**
## Description
This PR:
- adds a new dropdown to add users to multiple lists

### Things to note
We can currently add a user multiple times to a list... but I believe @jpmcb is working on a fix for that on the API side of things. Also, the UX is not completely at it yet, but we can have follow-ups to straighten things up with the UX.

## What type of PR is this? (check all applicable)

- [x] 🍕 Feature
- [ ] 🐛 Bug Fix
- [ ] 📝 Documentation Update
- [ ] 🎨 Style
- [ ] 🧑‍💻 Code Refactor
- [ ] 🔥 Performance Improvements
- [ ] ✅ Test
- [ ] 🤖 Build
- [ ] 🔁 CI
- [ ] 📦 Chore (Release)
- [ ] ⏩ Revert

## Related Tickets & Documents
closes #1828 

## Mobile & Desktop Screenshots/Recordings
image

## Added tests?

- [ ] 👍 yes
- [x] 🙅 no, because they aren't needed
- [ ] 🙋 no, because I need help

## Added to documentation?

- [ ] 📜 README.md
- [ ] 📓 docs.opensauced.pizza
- [ ] 🍕 dev.to/opensauced
- [ ] 📕 storybook
- [x] 🙅 no documentation needed

### CODE REVIEW

File: components/molecules/ContributorProfileHeader/contributor-profile-header.tsx [hunk 1]
```diff
@@ -23,9 +24,14 @@ import Text from "components/atoms/Typography/text";
 import { Textarea } from "components/atoms/Textarea/text-area";
 import { useUserCollaborations } from "lib/hooks/useUserCollaborations";
 import { useToast } from "lib/hooks/useToast";
-import { cardPageUrl } from "lib/utils/urls";
+// import { cardPageUrl } from "lib/utils/urls";
```
Comment (brandonroberts) [inline components/molecules/ContributorProfileHeader/contributor-profile-header.tsx:27]: nit(not blocking): Remove unused imports

```suggestion
```

File: components/molecules/ContributorProfileHeader/contributor-profile-header.tsx [hunk 2]
```diff
@@ -327,4 +329,79 @@ const ContributorProfileHeader = ({
...
+    const isOptionSelected = selectedList.some((s) => s.value === value.value);
+    if (isOptionSelected) {
+      setSelectedList((prev) => prev.filter((s) => s.value !== value.value));
+    } else {
+      setSelectedList((prev) => [...prev, value]);
+    }
+  };
+
+  const handleAddToList = async () => {
+    if (selectedList.length > 0 && contributor) {
```
Comment (brandonroberts) [inline components/molecules/ContributorProfileHeader/contributor-profile-header.tsx:352]: We could also fetch the contributor ID from GitHub if they are not in our data at all. This is good for now

File: lib/hooks/useList.ts [hunk 3]
```diff
@@ -69,6 +70,36 @@ const useFetchListContributors = (id: string, range = 30) => {
...
+      },
+    });
+
+    if (response.ok) {
+      const data = await response.json();
+      return {
+        data,
+        error: null,
+      };
+    }
```
Comment (brandonroberts) [inline lib/hooks/useList.ts:93]: Needs another return in case the response is not `ok` but doesn't throw an error

**Review by brandonroberts — CHANGES_REQUESTED** [review]

**Review by brandonroberts — APPROVED** [review]
Works! 🍕

**Review by jpmcb — APPROVED** [review]
🦖

**Pull request merged**